test/unit-test/cellular_at_core_utest.c

```diff
@@ -0,0 +1,932 @@
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    const char * pStr;
+    int32_t base;
+    int32_t * pResult;
+
+    /* Check that CELLULAR_AT_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATStrtoi( NULL, base, pResult );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+
+    /* Check that CELLULAR_AT_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATStrtoi( pStr, base, NULL );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+
+}
+
+/**
+ * @brief Test happy path for Cellular_ATStrtoi to return CELLULAR_AT_SUCCESS and got expected results.
+ */
+void test_Cellular_ATStrtoi_Happy_Path( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char * pStr = malloc( sizeof( char ) * ( strlen( CELLULAR_SAMPLE_STRTOL_HAPPY_CASE_STRING ) + 1 ) );
+    strcpy( pStr, CELLULAR_SAMPLE_STRTOL_HAPPY_CASE_STRING);
+    int32_t base = 10;
+    int32_t Result;
+
+    /* Check that CELLULAR_AT_SUCCESS is returned. */
+    cellularStatus = Cellular_ATStrtoi( pStr, base, &Result );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_SUCCESS, cellularStatus );
+    TEST_ASSERT_EQUAL( CELLULAR_SAMPLE_STRTOL_HAPPY_CASE_NUMBER, Result );
+    free( pStr );
+}
+
+/**
+ * @brief Test the error path for Cellular_ATStrtoi to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATStrtoi_Error_Path( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char * pStr = malloc( sizeof( char ) * ( strlen( CELLULAR_SAMPLE_STRTOL_ERROR_CASE_STRING ) + 1 ) );
+    strcpy( pStr, CELLULAR_SAMPLE_STRTOL_ERROR_CASE_STRING);
+    int32_t base = 10;
+    int32_t Result;
+
+    /* Check that CELLULAR_AT_ERROR is returned. */
+    cellularStatus = Cellular_ATStrtoi( pStr, base, &Result );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_ERROR, cellularStatus );
+    free(pStr);
+}
+
+/**
+ * @brief Test that any NULL parameter causes Cellular_ATStrDup to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATStrDup_Invalid_Param( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char * pDst;
+    char * pSrc;
+
+    /* Check that CELLULAR_AT_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATStrDup( NULL, pSrc );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+
+    /* Check that CELLULAR_AT_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATStrDup( pDst, NULL );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+
+}
+
+/**
+ * @brief Test happy path for Cellular_ATStrDup to return CELLULAR_AT_SUCCESS and got expected results.
+ */
+void test_Cellular_ATStrDup_Happy_Path( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char * pDst;
+    char pSrc[] = CELLULAR_SAMPLE_PREFIX_STRING_INPUT;
+
+    /* Check that CELLULAR_AT_SUCCESS is returned. */
+    cellularStatus = Cellular_ATStrDup( &pDst, pSrc );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_SUCCESS, cellularStatus );

```

Comment:
Verify that the string actually got duplicated by comparing the strings.

Reply:
Sure.